docs/knowledgebase/runtime/benchmarking.md

```diff
@@ -0,0 +1,547 @@
+    set_dummy_benchmark {
+      let b in 1 .. 1000;
+    }: {
+      set_dummy (RawOrigin::Root, b.into());
+    }
+    ```
+
+    The name of the benchmark case is `set_dummy_benchmark`. Here `b` is the input that passed into
+    the extrinsic `set_dummy`. `b` will be varied from 1 to 1,000 and run repeatedly for each value
+    of `b` according to the arguments we passed when running the benchmark command. We will get back
+    to this in the following.
+
+    The function `set_dummy` is called in the execution section, and we do not verify the result at
+    the end.
+
+    You will also see code like this:
+
+    ```rust
+    accumulate_dummy {
+      let b in 1 .. 1000;
+      let caller = account("caller", 0, 0);
+    }: _ (RawOrigin::Signed(caller), b.into())
+    ```
+
+    This is when the execution section is just a one-line code that call the extrinsic with the same
+    name as the benchmarking case name. Our `benchmark!` macro is smart enough to know that!
+
+4.
+    ```rust
+    benchmarks! {
+      // -- snip
+      impl_benchmark_test_suite!(Pallet, crate::tests::new_test_ext(),
+        crate::tests::Test);
+    }
+    ```
+
+    At the last line in the `benchmarks!` macro, `impl_benchmark_test_suite!` takes three input: the
+    module name, the function that generates a test genesis storage `new_text_ext()`, and a test
+    runtime we built when writing our test cases, and expand them into testing code.
+
+    Ultimately benchmarking code is expanded as testing code, so it takes the code you write in
+    testing (you do, right?) to build up [the mock
+    runtime](https://substrate.dev/docs/en/knowledgebase/runtime/tests#mock-runtime-environment). If
+    you are not familiar with how pallet test cases are written, please refer to our [Runtime Tests
+    chapter](tests).
+
+5. With all the code completed on the pallet side, you need to add code in the node runtime. First
+   add the necessary dependencies in [the runtime
+   `Cargo.toml`](https://github.com/paritytech/substrate/blob/master/bin/node/runtime/Cargo.toml).
+
+    ```toml
+    [dependencies]
+    # -- snip --
+    pallet-you-named = { version = ..., default-features = false, path = ... }
+
+    [features]
+    std = [
+      # -- snip --
+      "pallet-you-named/std"
+    ]
+    runtime-benchmarks = [
+      # -- snip --
+      "pallet-you-named/runtime-benchmarks"
+    ]
+    ```
+
+    Then in the node runtime, as usual, have your runtime implemements the pallet configurable trait
+    and add the pallet in `construct_runtime!`. For details, refer to [Add a Pallet to Your Runtime
+    Tutorial](../../tutorials/add-a-pallet). Then our runtime also implements
+    `frame_benchmarking::Benchmark` trait under a feature flag, and implement the
+    `dispatch_benchmark` method, as [shown in the
+    file](https://github.com/paritytech/substrate/blob/master/bin/node/runtime/src/lib.rs#L1412).
```

Comment:
```suggestion
    file](https://github.com/paritytech/substrate/blob/master/bin/node/runtime/src/lib.rs#L1412).
```

This link will become invalid any day as it is not bound to a commit, and only line number.

Reply:
updated.